### PR TITLE
NMS-15609: JDK17 Support

### PR DESCRIPTION
**Pull request by RangerRick**
This PR includes a number of updates to support JDK17:

* fix a number of integration tests to properly set `opennms.home` for JDK17's stricter `Paths.get(null)` handling
* handles JDK17's stricter controls on cross-module (and non-module) introspection (`--add-modules=`, `--add-reads=`, `--add-exports=`, and `--add-opens=`)
* bumps a few dependencies that needed to be updated to JDK17-compatible versions
* fixed a few other tests where issues cropped up from changes in type strictness or other small behavior changes
* change runtime docker images to use JDK17

During the development of this branch, it was also temporarily changed to perform a complete build/test/smoke run using only JDK17, so that any JDK17-specific bugs could be squashed.

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/NMS-15609

### CODE REVIEW

**Comment by j-white** [comment]
Nice! LGTM.

File: core/icmp-jna/src/test/java/org/opennms/jicmp/jna/Server.java
Comment (mershad-manesh) [inline core/icmp-jna/src/test/java/org/opennms/jicmp/jna/Server.java:1]: Minor item, should the copyright dates say 2023?

File: features/topology-map/plugins/org.opennms.features.topology.plugins.topo.asset/src/test/java/org/opennms/features/topology/plugins/topo/asset/filter/FilterParserTest.java
Comment (mershad-manesh) [inline features/topology-map/plugins/org.opennms.features.topology.plugins.topo.asset/src/test/java/org/opennms/features/topology/plugins/topo/asset/filter/FilterParserTest.java:1]: Minor item, should the copyright dates be 2023?

File: core/icmp-jna/src/test/java/org/opennms/jicmp/jna/Server.java [hunk 1]
```diff
@@ -1,8 +1,8 @@
 /*******************************************************************************
  * This file is part of OpenNMS(R).
  *
- * Copyright (C) 2011-2015 The OpenNMS Group, Inc.
- * OpenNMS(R) is Copyright (C) 1999-2015 The OpenNMS Group, Inc.
+ * Copyright (C) 2011-2022 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
```
Comment (christianpape) [inline core/icmp-jna/src/test/java/org/opennms/jicmp/jna/Server.java:5]:
```suggestion
 * Copyright (C) 2011-2023 The OpenNMS Group, Inc.
 * OpenNMS(R) is Copyright (C) 1999-2023 The OpenNMS Group, Inc.
```

File: features/topology-map/plugins/org.opennms.features.topology.plugins.topo.asset/src/test/java/org/opennms/features/topology/plugins/topo/asset/filter/FilterParserTest.java [hunk 2]
```diff
@@ -1,8 +1,8 @@
 /*******************************************************************************
  * This file is part of OpenNMS(R).
  *
- * Copyright (C) 2017-2017 The OpenNMS Group, Inc.
- * OpenNMS(R) is Copyright (C) 1999-2017 The OpenNMS Group, Inc.
+ * Copyright (C) 2017-2022 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
```
Comment (christianpape) [inline features/topology-map/plugins/org.opennms.features.topology.plugins.topo.asset/src/test/java/org/opennms/features/topology/plugins/topo/asset/filter/FilterParserTest.java:5]:
```suggestion
 * Copyright (C) 2017-2023 The OpenNMS Group, Inc.
 * OpenNMS(R) is Copyright (C) 1999-2023 The OpenNMS Group, Inc.
```

File: opennms-provision/opennms-provision-persistence/src/test/java/org/opennms/netmgt/provision/persist/policies/ScriptPolicyTest.java [hunk 3]
```diff
@@ -1,8 +1,8 @@
 /*******************************************************************************
  * This file is part of OpenNMS(R).
  *
- * Copyright (C) 2018 The OpenNMS Group, Inc.
- * OpenNMS(R) is Copyright (C) 1999-2018 The OpenNMS Group, Inc.
+ * Copyright (C) 2018-2022 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
```
Comment (christianpape) [inline opennms-provision/opennms-provision-persistence/src/test/java/org/opennms/netmgt/provision/persist/policies/ScriptPolicyTest.java:5]:
```suggestion
 * Copyright (C) 2018-2023 The OpenNMS Group, Inc.
 * OpenNMS(R) is Copyright (C) 1999-2023 The OpenNMS Group, Inc.
```

**Review by christianpape — CHANGES_REQUESTED** [review]
LGTM. Only a few headers need to be changed to 2023.

**Review by christianpape — APPROVED** [review]

**Review by christianpape — DISMISSED** [review]
LGTM.

File: core/cli/src/main/resources/bin/_module_opts.sh [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    "java.base/sun.net.www.protocol.http=ALL-UNNAMED" \
+    "java.base/sun.net.www.protocol.https=ALL-UNNAMED" \
+    "java.base/sun.net.www.protocol.jar=ALL-UNNAMED" \
+    "java.base/sun.net.www.content.text=ALL-UNNAMED" \
+    "java.base/sun.security.x509=ALL-UNNAMED" \
+    "jdk.xml.dom/org.w3c.dom.html=ALL-UNNAMED" \
+    "jdk.naming.rmi/com.sun.jndi.url.rmi=ALL-UNNAMED" \
+    "java.rmi/sun.rmi.registry=ALL-UNNAMED" \
+    "java.security.sasl/com.sun.security.sasl=ALL-UNNAMED" \
+    "org.apache.karaf.specs/org.apache.karaf.specs.locator=java.xml,ALL-UNNAMED"
```
Comment (j-white) [inline core/cli/src/main/resources/bin/_module_opts.sh:48]: This gets output on startup now: `WARNING: Unknown module: org.apache.karaf.specs specified to --add-exports`

**Review by mershad-manesh — APPROVED** [review]
The changes seems ok to me

**Comment by RangerRick** [comment]
Screw it, let's do this! Better to let any other problems get found in `develop` than languishing in a branch forever.

https://www.youtube.com/watch?v=PHFWpx1Y-_k

**Pull request merged**